### PR TITLE
[zstd] Update to version 1.5.6.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/zstd/fix-windows-rc-compile.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
+diff --git a/build/cmake/lib/CMakeLists.txt b/build/cmake/lib/CMakeLists.txt
+index 5d514ccb..dd79efb7 100644
+--- a/build/cmake/lib/CMakeLists.txt
++++ b/build/cmake/lib/CMakeLists.txt
+@@ -123,7 +123,7 @@ set(PUBLIC_INCLUDE_DIRS ${LIBRARY_DIR})
+ set(library_targets)
+ if (ZSTD_BUILD_SHARED)
+     add_library(libzstd_shared SHARED ${Sources} ${Headers} ${PlatformDependResources})
+-    target_include_directories(libzstd_shared INTERFACE $<BUILD_INTERFACE:${PUBLIC_INCLUDE_DIRS}>)
++    target_include_directories(libzstd_shared PUBLIC $<BUILD_INTERFACE:${PUBLIC_INCLUDE_DIRS}>)
```
Comment (dg0yt) [inline ports/zstd/fix-windows-rc-compile.patch:10]: I wonder what is this intented to achieve: There is not much public about BUILD_INTERFACE after install.
Same below.

Comment (teo-tsirpanis) [reply ports/zstd/fix-windows-rc-compile.patch:10]: See facebook/zstd#4009. On second thought I will change it only for the shared library.

Comment (dg0yt) [reply ports/zstd/fix-windows-rc-compile.patch:10]: Repeating in other words: does it put any directory into its PUBLIC (or INTERFACE) INSTALL_INTERFACE?

Comment (teo-tsirpanis) [reply ports/zstd/fix-windows-rc-compile.patch:10]: No. Its purpose is to pass the include directory to the resource compiler.

File: ports/zstd/usage [hunk 2]
```diff
@@ -1,4 +1,4 @@
 The package zstd provides CMake targets:
 
     find_package(zstd CONFIG REQUIRED)
-    target_link_libraries(main PRIVATE $<IF:$<TARGET_EXISTS:zstd::libzstd_shared>,zstd::libzstd_shared,zstd::libzstd_static>)
+    target_link_libraries(main PRIVATE zstd::libzstd)
```
Comment (dg0yt) [inline ports/zstd/usage:4]: Nit: Can we also align with current heuristics formatting:
~~~~
zstd provides CMake targets:

  find_package(zstd CONFIG REQUIRED)
  target_link_libraries(main PRIVATE zstd::libzstd)

Comment (teo-tsirpanis) [reply ports/zstd/usage]: Done.

**Comment by teo-tsirpanis** [comment]
CI fails when building `rtabmp` with this error:

```
ninja: error: 'D:/installed/x64-windows-static/lib/openblas.lib', needed by 'bin/RTABMap.exe', missing and no known rule to make it
```

**Comment by jimwang118** [comment]
All features passed with following triplets:

```
x64-windows-static
```

**Pull request merged**